### PR TITLE
Fix packaging tests on debian 10

### PR DESCRIPTION
**Pull request by alpar-t**
- lintian no longer has  `--fail-on-warnings`
- Systemd has a build number / version after it's acutal version
```
alpar@alpar-ci-debian-10-1015-1517:~$ systemd --version
systemd 241 (241)
+PAM +AUDIT +SELINUX +IMA +APPARMOR +SMACK +SYSVINIT +UTMP +LIBCRYPTSETUP +GCRYPT +GNUTLS +ACL +XZ +LZ4 +SECCOMP +BLKID +ELFUTILS +KMOD -IDN2 +IDN -PCRE2 default-hierarchy=hybrid
```

The link to the failing scan: https://gradle-enterprise.elastic.co/s/2pyo5k2zrvrta/tests/failed

Relates to #44657

### CODE REVIEW

**Comment by atorok** [comment]
@elasticmachine run elasticsearch-ci/packaging-matrix

**Comment by atorok** [comment]
@elasticmachine run elasticsearch-ci/bwc 
@elasticmachine run elasticsearch-ci/default-distro
@elasticmachine run elasticsearch-ci/packaging-matrix

File: qa/os/src/test/java/org/elasticsearch/packaging/test/PackageTests.java [hunk 1]
```diff
@@ -190,7 +190,7 @@ public void test50Remove() throws Exception {
             } else {
 
                 final Result versionResult = sh.run("systemctl --version");
-                final Matcher matcher = Pattern.compile("^systemd (\\d+)\n").matcher(versionResult.stdout);
+                final Matcher matcher = Pattern.compile("^systemd (\\d+)\\s").matcher(versionResult.stdout);
```
Comment (rjernst) [inline qa/os/src/test/java/org/elasticsearch/packaging/test/PackageTests.java]: Maybe just remove the newline? We just care about capturing the number in parens, and don't care about anything after that.

**Review by rjernst — APPROVED** [review]
LGTM. Thanks for tracking these down.

**Review by mark-vieira — APPROVED** [review]
LGTM

**Comment by atorok** [comment]
@elasticmachine run elasticsearch-ci/packaging-matrix

**Comment by atorok** [comment]
Debian packaging tests past PR, Centos 6 is failing to reach the repo.

**Pull request merged**